src/main/java/org/opensearch/performanceanalyzer/commons/os/ThreadDiskIO.java

```diff
@@ -0,0 +1,182 @@
+            this.avgPageCacheTotalThroughputBps = avgPageCacheTotalThroughputBps;
+        }
+
+        public String toString() {
+            return new StringBuilder()
+                    .append("rBps:")
+                    .append(avgReadThroughputBps)
+                    .append(" wBps:")
+                    .append(avgWriteThroughputBps)
+                    .append(" totBps:")
+                    .append(avgTotalThroughputBps)
+                    .append(" rSysc:")
+                    .append(avgReadSyscallRate)
+                    .append(" wSysc:")
+                    .append(avgWriteSyscallRate)
+                    .append(" totSysc:")
+                    .append(avgTotalSyscallRate)
+                    .append(" rPcBps:")
+                    .append(avgPageCacheReadThroughputBps)
+                    .append(" wPcBps:")
+                    .append(avgPageCacheWriteThroughputBps)
+                    .append(" totPcBps:")
+                    .append(avgPageCacheTotalThroughputBps)
+                    .toString();
+        }
+    }
+
+    private static void addSampleTid(String tid) {
+        try (FileReader fileReader =
+                        new FileReader(new File("/proc/" + pid + "/task/" + tid + "/io"));
+                BufferedReader bufferedReader = new BufferedReader(fileReader); ) {
+            String line = null;
+            Map<String, Long> kvmap = new HashMap<>();
+            while ((line = bufferedReader.readLine()) != null) {
+                String[] toks = line.split("[: ]+");
+                String key = toks[0];
+                long val = Long.parseLong(toks[1]);
+                kvmap.put(key, val);
+            }
+            tidKVMap.put(tid, kvmap);
+        } catch (FileNotFoundException e) {
+            LOGGER.debug("FileNotFound in parse with exception: {}", () -> e.toString());
+        } catch (Exception e) {
+            LOGGER.debug(
+                    "Error In addSample Tid for: {}  with error: {} with ExceptionCode: {}",
+                    () -> tid,
+                    () -> e.toString(),
+                    () -> StatExceptionCode.THREAD_IO_ERROR.toString());
+        }
+    }
+
+    public static synchronized void addSample() {
+        tids = OSGlobals.getTids();
+        oldtidKVMap.clear();
+        oldtidKVMap.putAll(tidKVMap);
+
+        tidKVMap.clear();
+        oldkvTimestamp = kvTimestamp;
+        kvTimestamp = System.currentTimeMillis();
+        for (String tid : tids) {
+            addSampleTid(tid);
+        }
+    }
+
+    public static synchronized void addSample(String threadId) {
+        tids = OSGlobals.getTids();
+        oldtidKVMap.clear();
+        oldtidKVMap.putAll(tidKVMap);
+
+        tidKVMap.clear();
+        oldkvTimestamp = kvTimestamp;
+        kvTimestamp = System.currentTimeMillis();
+        addSampleTid(threadId);
+        for (String tid : tids) {
+            addSampleTid(tid);
+        }
+    }
```

Comment:
Same comment as other place, instead of replicating code, let us refactor this to use method overloading. 

Also, move the Map clear and load part, timestamp update to the `private static void addSampleTid(String tid)`

Reply:
yeah - I just copy pasted the methods without too much thinking since my focus was mostly on the other side of the app - to make it workable and to be able to log the metrics for the given threadId. Makes sense

Reply:
Yeah this method was just copy paste it should be something like:
```java
    public static synchronized void addSample(String threadId) {
        tids = OSGlobals.getTids();
        oldtidKVMap.clear();
        oldtidKVMap.putAll(tidKVMap);

        tidKVMap.clear();
        oldkvTimestamp = kvTimestamp;
        kvTimestamp = System.currentTimeMillis();
        addSampleTid(threadId);
    }
```